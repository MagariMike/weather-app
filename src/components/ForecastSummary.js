import React from "react";
import iconData from "../data/iconData.json";

function ForecastSummary(props) {
  const { date, description, icon, temperature, onMoreDetailsClick } = props;
  const weatherCode = `${icon.slice(0, 1)}00`;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-summary" data-testid="forecast-summary">
      <div className="forecast-summary__date">{formattedDate}</div>
      <div className="forecast-summary__description">{description}</div>
      <div className="forecast-summary__icon" data-testid="forecast-icon">
        <img
          src={iconData[weatherCode]}
          alt="icon showing the current weather"
        />
      </div>
      <div className="forecast-summary__temperature">
        {temperature.max}&deg;C
      </div>
      <button onClick={onMoreDetailsClick} type="button">
        More Details
      </button>
    </div>
  );
}

export default ForecastSummary;
