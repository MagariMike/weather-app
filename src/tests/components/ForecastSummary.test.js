import React from "react";
import { render, screen } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 1111111,
    description: "Stub description",
    icon: "stubIcon",
    temperature: {
      min: 12,
      max: 22,
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );

    expect(screen.getByText("1111111")).toHaveAttribute(
      "class",
      "forecast-summary__date"
    );
    expect(screen.getByText("Stub description")).toHaveAttribute(
      "class",
      "forecast-summary__description"
    );
    expect(screen.getByText("stubIcon")).toHaveAttribute(
      "class",
      "forecast-summary__icon"
    );
    expect(screen.getByText("22Â°C")).toHaveAttribute(
      "class",
      "forecast-summary__temperature"
    );
  });
});
