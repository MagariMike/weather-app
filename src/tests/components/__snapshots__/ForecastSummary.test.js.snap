// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      class="forecast-summary__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-summary__description"
    >
      Stub description
    </div>
    <div
      class="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      <img
        alt="icon showing the current weather"
        src="https://openweathermap.org/img/wn/01d@2x.png"
      />
    </div>
    <div
      class="forecast-summary__temperature"
    >
      22Â°C
    </div>
    <button
      type="button"
    >
      More Details
    </button>
  </div>
</DocumentFragment>
`;
